app/src/main/java/com/github/se/signify/model/challenge/MockChallengeRepository.kt

```diff
@@ -0,0 +1,72 @@
+package com.github.se.signify.model.challenge
+
+class MockChallengeRepository : ChallengeRepository {
+  private val challenges = mutableMapOf<String, Challenge>()
+
+  var shouldSucceed: Boolean = true // Determines whether the operation succeeds
+  var exceptionToThrow: Exception = Exception("Simulated failure") // Exception for failures
+
+  var sendChallengeCalled: Boolean = false
+  var deleteChallengeCalled: Boolean = false
```

Comment:
I have implemented lists that track all the calls to the send/deleteChall. This may be overkill but it could prove useful. Now checking if the function has been called is a simple nonEmpty check.

Reply:
Overkill shouldn't be a problem in a mock like this. Nice work.